### PR TITLE
Memmove test suite

### PR DESCRIPTION
**Pull request by kouamdo**
`memmove` function shall return the pointer of the destination data

### CODE REVIEW

**Comment by kouamdo** [comment]
src should be null because after memmove , the data source should be free

File: src/tests/stdlib/memmove.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+EMBOX_TEST_SUITE("Test suite for memmove() function");
+
+
+TEST_CASE("move full data")
+{
+    const    char    src[]="string";
```
Comment (anton-bondarev) [inline src/tests/stdlib/memmove.c:11]: why can you not use char  src[]="string"; (without constant)?

File: src/tests/stdlib/memmove.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    char    dest[sizeof(src)+1];
+
+    test_assert_not_null(dest) ;
+    memmove((void*)dest,(void*)src,(size_t)strlen(src));
```
Comment (anton-bondarev) [inline src/tests/stdlib/memmove.c:15]: Too many casts

strlen returns size_t without casting
I think 
`memmove(dest, src, strlen(src));`
will enough

File: src/tests/stdlib/memmove.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+EMBOX_TEST_SUITE("Test suite for memmove() function");
+
+
+TEST_CASE("move full data")
+{
+    const    char    src[]="string";
+    char    dest[sizeof(src)+1];
+
+    test_assert_not_null(dest) ;
```
Comment (anton-bondarev) [inline src/tests/stdlib/memmove.c:14]: excess cheking

File: src/tests/stdlib/memmove.c (hunk 2 above)
Comment (kouamdo) [reply src/tests/stdlib/memmove.c:15]: ohhh sorry , i have done mistake ; thank you

File: src/tests/stdlib/memmove.c (hunk 3 above)
Comment (kouamdo) [reply src/tests/stdlib/memmove.c:14]: yes , to be very careful .... :) :) :+1:

File: src/tests/stdlib/memmove.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    test_assert_equal(*dest , *src) ;
+}
+
+
+TEST_CASE("check for NULL argument")
+{
+    const   char  src[]="\0";
+    char    dest[strlen(src)+1];
+
+    test_assert_not_null(dest) ;
```
Comment (anton-bondarev) [inline src/tests/stdlib/memmove.c:25]: excess checking
Remove all strings like
```
test_assert_not_null(dest) ;
```

**Pull request closed without merging**